### DISCUSSION TITLE
Difference in reducer behavior based on combine pullback order

### DISCUSSION THREAD

**Discussion by ferologics** [pointfreeco/swift-composable-architecture#535] [Q&A]
Hey, I'd like to ask if you could clarify this: what is the exact difference in TCA behavior when you combine & pullback a reducer **before** or **after** the "main" reducer?
```swift
foo = Reducer.combine (
  bar.pullback(...),
  Reducer { ... }
}
```
vs
```swift
foo = Reducer { ... }
  .combine(bar.pullback(...))
```
I assume the order of action execution depends on the provided reducer order? I.e. actions are first received by the first reducer, and then by any subsequently combined reducers? Is there some episode/case study explaining when combining in one order would be preferred over another? Or maybe it's easier to link me to the right place in the source code and this will become super obvious...

**Comment by andreyz** [comment]
Here's my rule of thumb list:
- Run local reducers before more global ones.
- Run optional reducers before the non-optional ones.

To learn all about composing reducers check the PointFree episode on [Higher-order reducers](https://www.pointfree.co/collections/composable-architecture/reducers-and-stores/ep71-composable-state-management-higher-order-reducers)

**Reply by ferologics** [reply]
This is also how I view it. I will re-watch the episode, thanks for sharing the link 👍

**Comment by stephencelis** [comment]
> I assume the order of action execution depends on the provided reducer order? I.e. actions are first received by the first reducer, and then by any subsequently combined reducers?

That's right! Reducers are run in order of combining, so mutations made by the first reducer are "seen" by the second. Effects are merged in the same order.
